### PR TITLE
ptl-tool: make branch name configurable

### PR DESCRIPTION
**Pull request by batrick**
Also, include hours/minutes in branch so multiple branches in a day have
different names.

Signed-off-by: Patrick Donnelly <pdonnell@redhat.com>

### CODE REVIEW

**Comment by joscollin** [comment]
> Also, include hours/minutes in branch so multiple branches in a day have
different names.

I thought of creating this as a PR after seeing it in `build-integration-branch` and then I forgot somehow :-)

File: src/script/ptl-tool.py [hunk 1]
```diff
@@ -88,11 +88,10 @@
 BASE_REMOTE = os.getenv("PTL_TOOL_BASE_REMOTE", "upstream")
 BASE_PATH = os.getenv("PTL_TOOL_BASE_PATH", "refs/remotes/upstream/heads/")
 GITDIR = os.getenv("PTL_TOOL_GITDIR", ".")
-USER = getpass.getuser()
+USER = os.getenv("PTL_TOOL_USER", getpass.getuser())
 with open(expanduser("~/.github.key")) as f:
     PASSWORD = f.read().strip()
-BRANCH_PREFIX = "wip-%s-testing-" % USER
-TESTING_BRANCH_NAME = BRANCH_PREFIX + datetime.datetime.now().strftime("%Y%m%d")
+TEST_BRANCH = os.getenv("PTL_TOOL_TEST_BRANCH", "wip-{user}-testing-%Y%m%d%H%M")
```
[the remark this answers is not in the document]
Comment (batrick) [reply src/script/ptl-tool.py]: > The new change overrides --merge-branch-name now

I don't think so?

````
$ src/script/ptl-tool.py --branch HEAD --merge-branch-name master 18192
Will merge PRs: [18192]
Detaching HEAD onto base: master
Merging PR #18192
Leaving HEAD detached; no branch anchors your commits
$ git log -1
commit 6fa6ccff4f195a5744b9a0419f9e98982a8c0d55 (HEAD)
Merge: bc139f02d2 d0732fc96f
Author: Patrick Donnelly <pdonnell@redhat.com>
Date:   Mon Oct 23 22:30:35 2017 -0700

    Merge PR #18192 into master
    
    * refs/pull/18192/head:
            qa/cephfs: test ec data pool
            qa/suites/fs/basic_functional/clusters: more osds
    
    Reviewed-by: Patrick Donnelly <pdonnell@redhat.com>
````

and without branch

````
$ src/script/ptl-tool.py --merge-branch-name master 18192
Will merge PRs: [18192]
Detaching HEAD onto base: master
Merging PR #18192
Checked out new branch wip-pdonnell-testing-20171023
Created tag testing/wip-pdonnell-testing-20171023
$ git log -1
commit 4c79bc7daf72a1f9153e78112ae8223326e25fd1 (HEAD -> wip-pdonnell-testing-20171023, tag: testing/wip-pdonnell-testing-20171023)
Merge: bc139f02d2 d0732fc96f
Author: Patrick Donnelly <pdonnell@redhat.com>
Date:   Mon Oct 23 22:32:03 2017 -0700

    Merge PR #18192 into master
    
    * refs/pull/18192/head:
            qa/cephfs: test ec data pool
            qa/suites/fs/basic_functional/clusters: more osds
    
    Reviewed-by: Patrick Donnelly <pdonnell@redhat.com>
````

[the remark this answers is not in the document]
Comment (batrick) [reply src/script/ptl-tool.py]: Whoops second example should be:

````
 src/script/ptl-tool.py --merge-branch-name master 18192
Will merge PRs: [18192]
Detaching HEAD onto base: master
Merging PR #18192
Checked out new branch wip-pdonnell-testing-201710240533
Created tag testing/wip-pdonnell-testing-201710240533
$ git log -1
commit 62f2ddb51eaf6f6b553d978a8ad764fb2e8f8bb4 (HEAD -> wip-pdonnell-testing-201710240533, tag: testing/wip-pdonnell-testing-201710240533)
Merge: bc139f02d2 d0732fc96f
Author: Patrick Donnelly <pdonnell@redhat.com>
Date:   Mon Oct 23 22:33:11 2017 -0700

    Merge PR #18192 into master
    
    * refs/pull/18192/head:
            qa/cephfs: test ec data pool
            qa/suites/fs/basic_functional/clusters: more osds
    
    Reviewed-by: Patrick Donnelly <pdonnell@redhat.com>
````

[the remark this answers is not in the document]
Comment (batrick) [reply src/script/ptl-tool.py]: Updated the PR to add seconds.

[the remark this answers is not in the document]
Comment (joscollin) [reply src/script/ptl-tool.py]: @batrick Could you please try this ? Is this desired behavior ?

$ export PTL_TOOL_USER=joscollin
$ src/script/ptl-tool.py --merge-branch-name wip-**jcollin**-testing 18192
Will merge PRs: [18192]
Detaching HEAD onto base: master
Merging PR #18192
Checked out new branch wip-**joscollin**-testing-20171024.065225
Created tag testing/wip-**joscollin**-testing-20171024.065225

[the remark this answers is not in the document]
Comment (batrick) [reply src/script/ptl-tool.py]: Yes, that looks right. I added a decimal point because it's difficult to read the date/time otherwise.

**Comment by batrick** [comment]
@joscollin looks good to merge?

**Comment by joscollin** [comment]
> looks good to merge?

@batrick Let me check once more.

File: src/script/ptl-tool.py [hunk 2]
```diff
@@ -293,16 +295,14 @@ def main():
     else:
         argv = sys.argv[1:]
     parser.add_argument('--branch', dest='branch', action='store', default=default_branch, help='branch to create ("HEAD" leaves HEAD detached; i.e. no branch is made)')
-    parser.add_argument('--merge-branch-name', dest='merge_branch_name', action='store', help='name of the branch for merge messages')
+    parser.add_argument('--merge-branch-name', dest='merge_branch_name', action='store', default=False, help='name of the branch for merge messages')
```
Comment (joscollin) [inline src/script/ptl-tool.py]: @batrick I don't understand what is the relevance of `--merge-branch-name` now. I think this option be removed, as branch name is configurable and automated by appending time to it. Am I missing something here ?  Please clarify. Thanks.

Comment (batrick) [reply src/script/ptl-tool.py]: It is useful in conjunction with `--branch HEAD`. I use it frequently when merging PRs to master:

`ptl-tool.py --branch HEAD --merge-branch-name master 12345`

Comment (batrick) [reply src/script/ptl-tool.py]: (A peculiarity of my setup is that I don' thave a local master branch because I find it useless.)

**Comment by joscollin** [comment]
@batrick The testing branch name is created using getuser() with no time appended, when I execute the script for the second time ?

```
jcollin@smithi042:~/test-ptl-18499$ echo $PTL_TOOL_USER
jos
jcollin@smithi042:~/test-ptl-18499$ src/script/ptl-tool.py --base master --pr-label wip-ptltool1-testing
Adding labeled PR #18284 to PR list
Will merge PRs: [18284]
Detaching HEAD onto base: master
Merging PR #18284
Checked out new branch wip-jos-testing-20171027.043558
Created tag testing/wip-jos-testing-20171027.043558
jcollin@smithi042:~/test-ptl-18499$ git branch
  18499
  master
* wip-jos-testing-20171027.043558
jcollin@smithi042:~/test-ptl-18499$ src/script/ptl-tool.py --base master --pr-label wip-ptltool1-testing
Adding labeled PR #18284 to PR list
Will merge PRs: [18284]
Detaching HEAD onto base: master
Merging PR #18284
Checked out new branch wip-jcollin-testing-20171027
Created tag testing/wip-jcollin-testing-20171027_03
jcollin@smithi042:~/test-ptl-18499$ git branch
  18499
  master
* wip-jcollin-testing-20171027
  wip-jos-testing-20171027.043558
```

**Comment by batrick** [comment]
@joscollin the issue is that you were using the `master` version of `ptl-tool.py` when you executed it the second time.

**Comment by joscollin** [comment]
@batrick Yes, that seems to be a mistake on my side. I'm not able to reproduce it now. :-)

> jcollin@smithi042:~/test-ptl-18499_10-31-2017$ export PTL_TOOL_USER=
> jcollin@smithi042:~/test-ptl-18499_10-31-2017$ src/script/ptl-tool.py --base master --pr-label wip-ptltool1-testing
> Adding labeled PR #18633 to PR list
> Will merge PRs: [18633]
> Detaching HEAD onto base: master
> Merging PR #18633
> [ 4343a215 ] BZ cited: Related-to: https://bugzilla.redhat.com/show_bug.cgi?id=684400
> [ https://github.com/ceph/ceph/pull/18633#pullrequestreview-73043634 ] Ceph tracker cited: http://tracker.ceph.com/issues/21000
> [ https://github.com/ceph/ceph/pull/18633#pullrequestreview-73049713 ] BZ cited: Related-to: https://bugzilla.redhat.com/show_bug.cgi?id=684400
> [ https://github.com/ceph/ceph/pull/18633#pullrequestreview-73049713 ] Ceph tracker cited: Fixes: http://tracker.ceph.com/issues/21210
> Need name for contributor "aniyanrajan"; Reviewed-by: Aniyan Rajan <aniyan.rajan@gmail.com>
> adding new contributors to githubmap on top of PR #18633
> Checked out new branch wip--testing-20171031.061659
> Created tag testing/wip--testing-20171031.061659

How about checking  `PTL_TOOL_USER` set to an empty value ? If it is empty, then let it take the username.

**Comment by batrick** [comment]
> How about checking PTL_TOOL_USER set to an empty value ? If it is empty, then let it take the username.

The right thing to do is to `unset PTL_TOOL_USER` in the calling shell.

**Comment by joscollin** [comment]
@batrick Please resolve the conflicts

**Comment by batrick** [comment]
done

**Pull request merged**